### PR TITLE
Api&UI registration

### PR DESCRIPTION
**Pull request by Kristinochka-bee**

### CODE REVIEW

File: src/test/java/pages/HomePage.java [hunk 1]
```diff
@@ -17,9 +17,9 @@ public class HomePage extends PageBase{
     public SelenideElement teacherSpotlightNonLoggedIn = $x("//*[@id='teacher-spotlight-non-logged-in']");
     public SelenideElement aboutUsButton = $x("//span[normalize-space()='About Us']");
     public SelenideElement coursesButton = $x("//span[normalize-space()='Courses']");
-    public SelenideElement ProfessorsButton = $x("//span[normalize-space()='Professors']");
+    public SelenideElement professorsButton = $x("//span[normalize-space()='Professors']");
     public SelenideElement signInButton = $x("//a[@id='sw-sign-in-submit-btn']");
-    public SelenideElement signUpButton = $x("//a[@id='sw-go-to-sign-up-btn']");
+    public SelenideElement signUpButton = $x("//div/a[@href='/sign-up']");
     public SelenideElement AddCourseButton = $x("//span[normalize-space()='Add course']");
```
Comment (LeonadO555) [inline src/test/java/pages/HomePage.java]: replace to addCourseButton

File: src/test/java/pages/HomePage.java [hunk 2]
```diff
@@ -17,9 +17,9 @@ public class HomePage extends PageBase{
...
     public SelenideElement CoursesDropDownMenu = $x("//span[normalize-space()='Add course']");
```
Comment (LeonadO555) [inline src/test/java/pages/HomePage.java]: replace to coursesDropDownMenu

File: src/test/java/pages/RegistrationPage.java [hunk 3]
```diff
@@ -3,32 +3,50 @@
...
+    public SelenideElement signInButton = $x("//a[@id='sw-go-to-sign-in-btn']");
 
-    public void selectRole(Roles role) {
-        selectOption(role.getValue(),roleOption);
-    }
-    public void userRegistration (UserEmails userEmail, Roles role){
-        selectRole(role);
+    public SelenideElement emptyRequiredFieldsMessageBlock= $x("//div[@class='error-message required-errors d-block']");
+    public SelenideElement existUserMessageBlock= $x("//div[@class='error-message signup-error d-flex']");
+    public void userRegistration ( Roles role, UserNames name, UserEmails userEmail){
```
Comment (LeonadO555) [inline src/test/java/pages/RegistrationPage.java]:
```
public void userRegistration ( Roles role, UserNames name, UserEmails userEmail, string password){
        selectRole(role, roleOption);
        fullNameInput.sendKeys(name);
        emailInput.sendKeys(userEmail);
        passwordInput.sendKeys(password);
        checkBox.click();
        singUpButton.click();
    }
```

File: src/test/java/api/UserApi.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        dto = new RegistrationDTO();
+        dto.setFull_name(faker.name().username());
+        dto.setEmail(faker.internet().safeEmailAddress());
+        dto.setPassword("123456");
+        dto.setGenerate_magic_link("false");
+        return dto;
+    }
+    public Response registrationNewUserApi(Integer code, RegistrationDTO randomDataForRegisterUser){
+        String endpoint = "/v1/api/users";
+        String name = randomDataForNewUser().getFull_name();
```
Comment (LeonadO555) [inline src/test/java/api/UserApi.java]: remove

File: src/test/java/api/UserApi.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+        response = doPostRequest(endpoint, code, randomDataForRegisterUser);
```
Comment (LeonadO555) [inline src/test/java/api/UserApi.java]: replace randomDataForRegisterUser to randomDataForNewUser()

File: src/test/java/api/UserApi.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+    public RegistrationDTO randomDataForNewUser(){
+        dto = new RegistrationDTO();
+        dto.setFull_name(faker.name().username());
+        dto.setEmail(faker.internet().safeEmailAddress());
+        dto.setPassword("123456");
+        dto.setGenerate_magic_link("false");
+        return dto;
+    }
+    public Response registrationNewUserApi(Integer code, RegistrationDTO randomDataForRegisterUser){
```
Comment (LeonadO555) [inline src/test/java/api/UserApi.java]: remove RegistrationDTO randomDataForRegisterUser

File: src/test/java/tests/e2e/RegistrationNewUserUITest.java [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+    public void registrationNewUserViaApiCheckViaUIAndDeleteViaApiTest(){
+        userApi = new UserApi();
+        RegistrationDTO registrationDTO = userApi.randomDataForNewUser();
+        Response response = userApi.registrationNewUserApi(201, registrationDTO);
+        String userEmail = response.jsonPath().getString("email");
+        String userName = response.jsonPath().getString("full_name");
+        homePage = new HomePage();
+        homePage.avatarButton.isDisplayed();
+        homePage.professorsButton.click();
+        professorSpotlightPage = new ProfessorSpotlightPage();
```
Comment (LeonadO555) [inline src/test/java/tests/e2e/RegistrationNewUserUITest.java:38]: Add check display email

File: src/test/java/tests/api/RegistrationAndDeleteUserApiTest.java [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+        userApi.registrationNewUserApi(200, registrationDTO);
+    }
+
+    @Test
+    public void registrationNewUserAndDeleteApi(){
+        userApi = new UserApi();
+        RegistrationDTO registrationDTO = userApi.randomDataForNewUser();
+        Response response = userApi.registrationNewUserApi(201, registrationDTO);
+        String userEmail = response.jsonPath().getString("email");
+        userApi.deleteExistingUser(userEmail, 200);
```
Comment (LeonadO555) [inline src/test/java/tests/api/RegistrationAndDeleteUserApiTest.java:18]: add check that user is delete

File: src/test/java/pages/RegistrationPage.java (hunk 3 above)
Comment (Kristinochka-bee) [reply src/test/java/pages/RegistrationPage.java]: fixed

File: src/test/java/api/UserApi.java (hunk 4 above)
Comment (Kristinochka-bee) [reply src/test/java/api/UserApi.java]: done

File: src/test/java/api/UserApi.java (hunk 5 above)
Comment (Kristinochka-bee) [reply src/test/java/api/UserApi.java]: done

File: src/test/java/pages/HomePage.java (hunk 2 above)
Comment (Kristinochka-bee) [reply src/test/java/pages/HomePage.java]: fixed

File: src/test/java/pages/HomePage.java (hunk 1 above)
Comment (Kristinochka-bee) [reply src/test/java/pages/HomePage.java]: fixed

File: src/test/java/api/UserApi.java (hunk 6 above)
Comment (Kristinochka-bee) [reply src/test/java/api/UserApi.java]: done

File: src/test/java/tests/api/RegistrationAndDeleteUserApiTest.java (hunk 8 above)
Comment (Kristinochka-bee) [reply src/test/java/tests/api/RegistrationAndDeleteUserApiTest.java:18]: done

File: src/test/java/tests/e2e/RegistrationNewUserUITest.java (hunk 7 above)
Comment (Kristinochka-bee) [reply src/test/java/tests/e2e/RegistrationNewUserUITest.java:38]: done

**Review by LeonadO555 — APPROVED** [review]

**Pull request merged**